### DISCUSSION TITLE
Patch and Queue based training: how is step per epoch calculated

### DISCUSSION THREAD

**Discussion by Paddy-Xu** [TorchIO-project/torchio#1010] [Q&A]
Hi, I am using the Patch based training. However, it seems that the steps_per_epoch is calculated very weirdly. Or equivalently, how is the number of training patches calculated?

Suppose I have only one single training image of 1000x1000x1000. Using a patch size of 100 and a batch size of 10, I would assume the step_per_epoch is a least (1000/100)^3/10 = 100. However, it seems that each epoch contains only one batch. Or equivalently, how is the data length calculated AFTER generating patches? In this case, I would assume it is at least (1000/100)^3 = 1000 

Although in theory, this would not affect the performance, and I could just set a ```check_val_every_n_epoch```, I am wondering if it possible to solve the problem without setting an interval between validation steps.

Thanks in advance!

**Comment by fepegar** [comment]
Hi, @Paddy-Xu. Apologies for the delayed response. If you can provide a minimal working example wo we can understand the issue, we'll be able to help with your question.
